### ISSUE TITLE
False positive for INT_BAD_COMPARISON_WITH_SIGNED_BYTE

### ISSUE THREAD

**Issue by nmatt** [spotbugs/spotbugs#4192]
**Describe the bug**
Testing that a byte value is strictly less than 127 produces "Low: Bad comparison of signed byte with 127", although that's a valid comparison.

**To Reproduce**
```java
public class Bug
{
    public static boolean isPrintable(byte b)
    {
        return b > 32 && b < 127;
    }
}
```

**Expected behavior**
Spotbugs should not report anything for `b < 127`. Only `b <= 127` or `b < 128` should trigger a diagnostic.

**Environment**
- Java Compiler: Eclipse Temurin
- Java version: JDK 17
- SpotBugs Version: 4.10.2
- How do you use SpotBugs: Maven Plugin
    - Plugin version: 4.10.2.0
    - Environment version for Plugin: Maven 3.8.4
